zdigness/puttdle
# [TASK] Server: Add Github actions, unit tests, env vars, update docs

## Description
The following is added/updated in this PR:
1. Github workflows
2. env example file
3. env vars with DB config
4. Port for Postgres DB from "5438:5432" -> "5432:5432"
5. Rename files to be specific to what they are and remove redundancies 
6. `Console.log` -> `Console.error`
7. Database seeding


## Motivation and Context
In this PR, my goal was to setup Github actions to validate we are passing the following standards before being able to merge
1. eslint
2. Code coverage with testing
3. Docker-compose

I also added env vars to protect any secrets.

Unit test have been added to harden and verify the work going in is validated and continues to be so as the project go on

## How Has This Been Tested?

Unit tests
1.  `npm run test` in server directory

Docker Compose
1. See steps in `server/README.md`
2. View artifacts

## Artifacts (if appropriate):


### Running Server Locally With Docker DB
https://github.com/zdigness/puttdle/assets/23369330/c1e1ce6c-39d4-42d0-9e1e-e7536b86c771

### Docker Compose With Env Config

https://github.com/zdigness/puttdle/assets/23369330/0a13425b-0422-468f-aaad-5ad061955fef

### Jest Unit Tests

![image](https://github.com/zdigness/puttdle/assets/23369330/818547ba-3868-434b-b602-6c93855fd839)
